backend/src/test/java/de/mightypc/backend/service/hardware/PcCaseServiceTest.java

```diff
@@ -0,0 +1,167 @@
+package de.mightypc.backend.service.hardware;
+
+import de.mightypc.backend.model.specs.PcCase;
+import de.mightypc.backend.repository.hardware.PcCaseRepository;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+class PcCaseServiceTest {
+    private final PcCaseRepository pcCaseRepository = mock(PcCaseRepository.class);
+    private final PcCaseService pcCaseService = new PcCaseService(pcCaseRepository);
+
+    @Test
+    void getAll_ReturnsAllPcCases() {
+        // Arrange
+        List<PcCase> expected = new ArrayList<>(List.of(new PcCase("1", "test", 1.01f, 1.01f)));
+        when(pcCaseRepository.findAll()).thenReturn(expected);
+
+        // Act
+        List<PcCase> actual = pcCaseService.getAll();
+
+        // Assert
+        assertEquals(expected, actual, "getAll should return all PcCases");
+        verify(pcCaseRepository).findAll();
+    }
+
+    @Test
+    void getAll_ReturnsEmptyListWhenNoPcCases() {
+        // Arrange
+        List<PcCase> expected = new ArrayList<>();
+        when(pcCaseRepository.findAll()).thenReturn(expected);
+
+        // Act
+        List<PcCase> actual = pcCaseService.getAll();
+
+        // Assert
+        assertEquals(expected, actual, "getAll should return an empty list when no PcCases exist");
+        verify(pcCaseRepository).findAll();
+    }
+
+    @Test
+    void getById_ReturnsPcCaseWhenExists() {
+        //Arrange
+        Optional<PcCase> expected = Optional.of(new PcCase("1", "test", 1.01f, 1.01f));
+        when(pcCaseRepository.findById("1")).thenReturn(expected);
+
+        //Act
+        Optional<PcCase> actual = pcCaseService.getById("1");
```

Comment:
Hast du auch einen Test case, falls die Id sich nicht in der Datenbank befindet?